File: Src/App/zabotalaboratory/zabotalaboratory.Analyses.Database/Entities/AnalysesTypes.cs
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using zabotalaboratory.Common.Datamodel.Abstractions;

namespace zabotalaboratory.Analyses.Database.Entities
{
    public class AnalysesTypes : IValidatable
    {
        public virtual int Id { get; set; }

        public virtual int NumberInOrder { get; set; }

        [Required]
        public virtual string Name { get; set; }

        public virtual string ExcelName { get; set; }

        public virtual string PDFName { get; set; }

        [Required]
        public virtual int Number1C { get; set; }

        [Required]
        public virtual bool IsValid { get; set; }
        
        public virtual List<LaboratoryAnalysesTests> LaboratoryAnalysesTests { get; set; }

        public virtual string BioMaterial { get; set; }
    }
}

File: Src/App/zabotalaboratory_client/src/app/shared/models/analyses/analyses-types.ts
import {LaboratoryAnalysesTests} from "./laboratory-analyses-tests";

export class AnalysesTypes {
  public id: number;
  public number1C: number;
  public numberInOrder: number;
  public name: string;
  public excelName: string;
  public pdfName: string;
  public isValid: boolean;
  public laboratoryAnalysesTests: LaboratoryAnalysesTests[];
  public bioMaterial: string;
}
